docs/src/views/CustomComponents.tsx

```diff
@@ -0,0 +1,234 @@
+import * as React from 'react'
+import * as cx from 'classnames'
+import { NavLink } from 'react-router-dom'
+import { Header } from 'semantic-ui-react'
+import {
+  Button,
+  Divider,
+  Provider,
+  createComponent,
+  ComponentSlotStyle,
+  ComponentVariablesInput,
+} from '@stardust-ui/react'
+
+import DocPage from '../components/DocPage/DocPage'
+import ExampleSnippet from '../components/ExampleSnippet/ExampleSnippet'
+
+interface StyledButtonProps {
+  className?: string
+  styles?: ComponentSlotStyle
+  variables?: ComponentVariablesInput
+  children?: React.ReactNodeArray | React.ReactNode
+}
+
+const StyledButton: React.SFC<StyledButtonProps> = createComponent<StyledButtonProps, {}>({
+  displayName: 'StyledButton',
+  render({ stardust, className, children }) {
+    const { classes } = stardust
+    return <button className={cx(className, classes.root)}>{children}</button>
+  },
+})
+
+export default () => (
+  <DocPage title="Custom Components">
+    <Header as="h2" content="Overview" />
+    <p>
+      You can use your own components as part of the Stardust's theming mechanism. In order for all
+      theming aspects to be available to your custom components, you should use the{' '}
+      <code>createComponent</code>
+      function, provided by the Stardust library.
+    </p>
+
+    <Header as="h2" content="Create custom component" />
+    <p>
+      Let's take a look into one simple example of using the <code>createComponent</code> function
+      for adapting your custom component to the Stardust's theming mechanism.
+    </p>
+
+    <ExampleSnippet
+      value={[
+        `import { createComponent } from '@stardust-ui/react'`,
+        ``,
+        `export interface StyledButtonProps {`,
```

Comment:
I would suggest to remove TS from this example - main reason is that it introduces quite a lot of 'noise', and additional one (and the one that applies to broader community) is that our docs should be readable for folks that have no prior knowledge of TS